tools/CAT/src/app/features/pages/pageExtraRecuders.ts

```diff
@@ -0,0 +1,133 @@
+import { addPage, updatePage, deletePage } from './pageThunks';
+import { PageDto, CategoryDto, SubCategoryDto } from '@/models/api';
+
+const extraReducersConfigPages = (builder) => {
+	//add
+	builder.addCase(addPage.pending, (state) => {
+		state.loading = true;
+		state.error = null;
+	});
+	builder.addCase(addPage.fulfilled, (state, action) => {
+		const { subCategoryId } = action.payload;
+
+		const updatedCategories = state.categories?.map(
+			(category: CategoryDto) => {
+				const updatedSubCategories = category.subCategories?.map(
```

Comment:
I would suggest to use here findIndex or find methods instead of mapping, it would remove unnecessary mapping and optimize the code